code/modules/turbolift/turbolift_machines.dm

```diff
@@ -0,0 +1,300 @@
+	anchored = TRUE
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF | FREEZE_PROOF
+	verb_say = "beeps"
+	verb_ask = "beeps"
+	verb_exclaim = "beeps"
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 2
+
+	var/shuttle_id //Needs to match the turbolift computer & mobile dock
+	var/floor_id
+
+/obj/machinery/turbolift_button/Initialize()
+	. = ..()
+	if(!shuttle_id)
+		log_mapping("TURBOLIFT: [src] has no shuttle_id at [AREACOORD(src)]")
+		message_admins("TURBOLIFT: [src] has no shuttle_id at [AREACOORD(src)]")
+		return
+	floor_id = "[shuttle_id]_[src.z]"
+
+/obj/machinery/turbolift_button/attack_hand(mob/user)
+	if (stat & NOPOWER)
+		to_chat(user, "<span class='warning'>[src] does not respond.</span>")
+	if(!shuttle_id || !floor_id)
+		say("An unexpected error has occured. Please contact a Nanotrasen Turbolift Repair Technician.")
+		return
+
+	var/obj/docking_port/mobile/turbolift/M = SSshuttle.getShuttle(shuttle_id)
+	var/obj/machinery/computer/turbolift/T = M?.turbolift_computer?.resolve()
+	if(!M || !T)
+		say("An unexpected error has occured. Please contact a Nanotrasen Turbolift Repair Technician.")
+		return
+
+	if("[floor_id]" in T.destination_queue)
+		say("The current deck is already queued.")
+	else if(T.z == src.z)
+		say("The turbolift is already at this deck.")
+	else
+		say("The turbolift will arrive shortly. Thank you for using Nanotrasen Turbolift Services(TM).")
+		T.destination_queue += "[floor_id]"
+		START_PROCESSING(SSmachines, T)
+
+/obj/machinery/computer/turbolift
+	name = "turbolift control console"
+	icon = 'icons/obj/turbolift.dmi'
+	icon_state = "panel"
+	density = FALSE
+	anchored = TRUE
+	can_be_unanchored = FALSE
+	mouse_over_pointer = MOUSE_HAND_POINTER
+	desc = "Nanotrasen's decision to replace the iconic turboladder was not met with unanimous praise, experts citing increased obesity figures from crewmen no longer needing to climb vertically through several miles of deck to reach their target. However this is undoubtedly much faster."
+	pixel_y = 32
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF | FREEZE_PROOF
+
+	var/shuttle_id //Needs to match the mobile docking port's ID
+	var/list/possible_destinations = list()
+	var/list/airlocks = list()
+	var/list/destination_queue = list()
+	var/time_between_stops = 300 //Time in deciseconds before going to the next location in the queue.
+	var/in_use = FALSE
+	var/online = TRUE //Is the elevator functional? Will be expanded upon later
+
+
+/obj/machinery/computer/turbolift/Initialize()
+	. = ..()
+	GLOB.turbolifts += src
+
+/obj/machinery/door/airlock/turbolift
+	name = "turbolift airlock"
+	icon = 'icons/obj/turbolift_door.dmi'
+	desc = "A sleek airlock for walking through. This one looks extremely strong."
+	icon_state = "closed"
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF | FREEZE_PROOF
+	var/dock_dir
+
+/obj/machinery/door/airlock/bolt()
```

Comment:
What're these for

Reply:
Subtype that doesn't play audio or send messages when bolting/unbolting so the player isn't spammed with 6 messages and soundfiles playing at once.

Reply:
Shouldn't this be an elevator specific thing then??

Reply:
oops I goofed the path

thanks copypasta